.pre-commit-config.yaml

```diff
@@ -89,28 +89,26 @@ repos:
         always_run: false
         pass_filenames: false
         types: [sql]
-        additional_dependencies:
-          - dbt-snowflake==1.5.1
-          - dbt-core==1.5.2
-  # Note: for SQLFluff we override the entrypoint for two reasons:
+  # Note: for SQLFluff we don't use the default pre-commit hook for a few reasons:
   #   1. dbt cloud looks for config in the dbt project directory, so we put it in
   #      transform/, however sqlfluff requires the templater config to be in the
   #      current working directory. So we have to cd to transform/ to properly set the
   #      templater config.
   #   2. Some machines (such as ODI Macs) seem to struggle with sqlfluff's
   #      multiprocessing implementation, so we force it to be single-process.
-  - repo: https://github.com/sqlfluff/sqlfluff
-    rev: 2.1.2
+  #   3. The pre-commit managed python environment can be difficult to install,
+  #      especially due to issues with pyarrow being brought in by Snowflake.
+  #      This keep things more predictable by using the poetry.lock environment.
+  - repo: local
     hooks:
-      - id: sqlfluff-lint
-        entry: bash -c 'cd transform && sqlfluff lint --processes 1'
-        additional_dependencies:
-          - dbt-snowflake==1.5.1
-          - sqlfluff-templater-dbt==2.1.2
-          - dbt-core==1.5.2
-      - id: sqlfluff-fix
-        entry: bash -c 'cd transform && sqlfluff fix --processes 1'
-        additional_dependencies:
-          - dbt-snowflake==1.5.1
-          - sqlfluff-templater-dbt==2.1.2
-          - dbt-core==1.5.2
+      - id: sqlfluff
+        name: sqlfluff
+        language: python
+        description: "Lints sql files with `SQLFluff`"
+        types: [sql]
+        require_serial: true
+        entry: >
+          poetry run python -c "import os, subprocess; os.chdir('transform');
```

Comment:
This is an ugly hack, but the reason for this is to not assume that `bash` or `sh` are installed on the user's system. We *know* python and its stdlib are installed, so it's safe to change directories using that.